Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Material/MaterialPipelineSourceData.h

```diff
@@ -63,7 +64,7 @@ namespace AZStd
     {
         size_t operator()(const AZ::RPI::MaterialPipelineSourceData::ShaderTemplate& value) const
         {
-            return hash<AZStd::string>()(value.m_shader) ^ hash<AZStd::string>()(value.m_azsli);
+            return hash<AZStd::string>()(value.m_shader) ^ hash<AZStd::string>()(value.m_preMaterialAzsli) ^ hash<AZStd::string>()(value.m_postMaterialAzsli);
```

Comment:
Nit: can leave as is but using `xor` as a hash combining operator can get you into trouble, so it'd be nice if we had a more "official" mechanism for this (if only to avoid the cognitive overhead of needing to determine each time we review the code if the hash combination is safe). In particular, `xor` is a symmetric operation, so if we exchanged the values of any of these fields, the resulting hash would be the same (again, doesn't matter here, but this is just an example of one of the pitfalls to be aware of).

Reply:
I reverted this change.